Development byunghoon

Harvey-JJ opened the pull request:
[Feature] Employee Class 추가 건입니다.
Employee Class는 Employee 정보를 저장하는 역할을 하며
Key값에 맞는 Value를 Return하는 getValue(String Key) 함수를 포함하고 있습니다.

Harvey-JJ committed:
commit test

commit test

Harvey-JJ committed:
Employee Class 추가

Employee 정보를 저장하고 필요한 정보를 Return 하는 Class 추가

Harvey-JJ committed:
Employee Test Code 추가

Employ의 getValue 함수 Test Code 추가

Harvey-JJ committed:
Revert

Revert

Rakjunyang commented on EmployeeManager/src/test/java/EmployeeTest.java:
```diff
@@ -0,0 +1,72 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class EmployeeTest {
```
모든 변수들에 대해서 get 함수가 정상 동작하는지 확인 할 수 있도록 Test Case 들 잘 작성하신 것 같습니다!
이 Test 코드가 있으면 이후 개발도 bug 없이 잘 진행 할 수 있을 것 같습니다.

Rakjunyang commented on EmployeeManager/src/main/java/Employee.java:
```diff
@@ -0,0 +1,69 @@
+        String cl, String phoneNumber, String firstNumber, String midNumber,
+        String lastNumber, String birthDay, String certi) {
+        this.id = id;
+        this.sortKey = Integer.parseInt(id.substring(0, 2)) < 30 ? 2000000000 + Integer.parseInt(id) :  1900000000 + Integer.parseInt(id);
```
이런 변수가 있으면, 강사님께서도 강조하셨던 입사년도 , 사번 정렬 문제를 한번에 해결 할 수 있을것 같네요!

Rakjunyang commented on EmployeeManager/src/main/java/Employee.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+
+    public String getValue(String key){
```
각 변수 마다 별도의 getter 함수를 만들지 않아 깔끔해서 좋은 것 같습니다.
향후에는 Factory Method 패턴이나 다른 방법을 활용해서 if 문들을 함수에서 빼내는 것도 좋을 것 같아요~

preventionYun commented on EmployeeManager/src/main/java/Employee.java:
```diff
@@ -0,0 +1,69 @@
+        if(key == "birthDay"){
+            return birthDay;
+        }
+        return certi;
```
여기에는 추후에 key가 잘못 들어오는 케이스에는 certi가 리턴될 수도 있을 것 같습니다! certi까지 키 값 체크하고 아닌 경우에 에러처리할 수 있도록 변경하는 것은 어떨까요??

Harvey-JJ replied:
좋은 아이디어 감사드립니다! 추가 수정 하도록 하겠습니다~

Harvey-JJ committed:
[Refactoring] 변수 변경 및 함수 수정

1. 지윤님께서 리뷰해주신대로 key 값이 없을 경우 null을 return하도록 수정했습니다.
2. set 함수를 추가했습니다.
3. 옵션에 따른 검색 key 값을 employee에서 가지고 있게 되면 option 추가시마다 employee 수정이 필요하여 해당 부분을 option selector에 이관하기 위해 일부 column을 제거했습니다.

preventionYun commented:
PR 제목은 앞에 prefix를 붙여주면 어떤 작업을 하셨는지 보기 좋을 것 같아요! 

[feature] Employee class 추가

이런식으로 변경하는 것은 어떨까요?

아래 Ground rule에 있는 내용입니다! 

feature : 새로운 기능 추가
fix : 버그 수정
docs : 문서 수정
test : 테스트 코드 추가
refactor : 코드 리팩토링
style : 코드 의미에 영향을 주지 않는 변경사항
chore : 빌드 부분 혹은 패키지 매니저 수정사항

Harvey-JJ committed:
[Refactoring] 불필요 테스트 삭제

Employee에 변수 삭제에 따른 테스트 코드 수정입니다.

Harvey-JJ committed:
[Refactoring] 생성자 변경

생성자에 불필요한 인자 제거

Harvey-JJ committed:
[fix] 함수 인자 bug-fiux

함수 인자 bug-fiux
